operatorcontrolplane/v1alpha1/types_conditioncheck.go

```diff
@@ -0,0 +1,150 @@
+// Package v1alpha1 is an API version in the controlplane.operator.openshift.io group
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PodNetworkConnectivityCheck
+type PodNetworkConnectivityCheck struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	// Spec defines the source and target of the connetivity check
+	// +kubebuilder:validation:Required
+	// +required
+	Spec PodNetworkConnectivityCheckSpec `json:"spec"`
+
+	// Status contains the observed status of the connectivity check
+	// +optional
+	Status PodNetworkConnectivityCheckStatus `json:"status"`
+}
+
+type PodNetworkConnectivityCheckSpec struct {
+	// SourcePod names the pod from which the condition will be checked
+	// +kubebuilder:validation:Required
+	// +required
+	SourcePod string `json:"sourcePod"`
+
+	// EndpointAddress to check. A TCP address of the form host:port. Note that
+	// if host is a DNS name, then the check would fail if the DNS name cannot
+	// be resolved. Specify an IP address for host to bypass DNS name lookup.
+	// +kubebuilder:validation:Required
+	// +required
+	TargetEndpoint string `json:"targetEndpoint"`
```

Comment:
set the format, compare `kubebuilder:validation:Format` and https://github.com/kubernetes/apiextensions-apiserver/blob/master/pkg/apis/apiextensions/v1/types_jsonschema.go#L33

Reply:
I didn't see an appropriate format (e.g. host:port), so I will use a pattern instead:
```
// +kubebuilder:validation:Pattern=^\S+:\d*$
``` 
(one or more non-whitespace + ':' + port number )

Reply:
looks fine

Reply:
am not fluent in regexes enough: `\S` includes numbers and other chars allowed in hosts?

Reply:
`\S` is non-whitespace. So the regex means anything-non-whitespace, colon, digits.

Reply:
@sttts The regex isn't perfect, I started to write a more thorough one, but didn't see the point if it wasn't going to be perfect as it would have to handle hostnames, ipv4, ipv6 (which has :, and %, and brackets, etc..) simultaneously.